rules-reviewed/eap7/eap6/hsearch.windup.xml

```diff
@@ -0,0 +1,899 @@
+            <perform>
+                <lineitem message="Maven artifact org.hibernate:hibernate-search-analyzers was removed from Hibernate Search 5. We recommend to depend on the appropriate Lucene artifact directly, for example org.apache.lucene:lucene-analyzers-common." />
+            </perform>
+        </rule>
+        <rule id="hsearch-00201">
+            <when>
+                <xmlfile in="pom.xml" matches="//*[local-name() = 'artifactId' and text() = 'hibernate-search-analyzers']" />
+            </when>
+            <perform>
+                <hint title="hibernate-search-analyzers maven artifact dropped in Hibernate Search 5" effort="1" severity="mandatory">
+                    <message>The hibernate-search-analyzers module was removed. The recommended approach is to directly use the appropriate Lucene artifact,
+                     for example org.apache.lucene:lucene-analyzers-common.</message>
+                    <link
+                        href="https://access.qa.redhat.com/documentation/en/red-hat-jboss-enterprise-application-platform/version-7.0/migration-guide/#migrate_miscellaneous_hibernate_search_changes"
+                        title="Miscelanous Changes in Hibernate Search 5.x" />
+                    <tag>hibernate-search</tag>
+                    <tag>maven</tag>
+                </hint>
+            </perform>
+        </rule>        
+        <rule id="hsearch-00210">
+            <when>
+                <javaclass references="org.apache.lucene.search.SortField({*})">
+                    <location>CONSTRUCTOR_CALL</location>
+                </javaclass>
+            </when>
+            <perform>
+                <hint title="Lucene API changed for org.apache.lucene.search.SortField constructor" effort="1" severity="mandatory">
+                    <message>
+                    The Lucene SortField API requires the following application code change. In the previous release of JBoss EAP, you set the type of the sort field in the query as follows.
+
+```java
+fulltextQuery.setSort(new Sort(new SortField("title", SortField.STRING)));
+```
+
+The following is an example of how you set it in JBoss EAP 7.
+
+```java
+fulltextQuery.setSort(new Sort(new SortField("title", SortField.Type.STRING)))
+```
+
+                     </message>
+                    <link href="http://hibernate.org/search/documentation/migrate/5.5/#api-changes" title="Hibernate Search 5.4 to 5.5 migration" />
+                    <link
+                        href="https://access.qa.redhat.com/documentation/en/red-hat-jboss-enterprise-application-platform/version-7.0/migration-guide/#migrate_miscellaneous_hibernate_search_changes"
+                        title="Miscellaneous Hibernate Search Changes" />
+                </hint>
+
+            </perform>
+        </rule>
+        <rule id="hsearch-00211">
+            <when>
+                <javaclass references="org.hibernate.search.cfg.ContainedInMapping.numericField{*}">
```

Comment:
Windup will catch cases like that if the method chain is properly imported and the appropriate types are in the analysis classpath.